internal/content/graph.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package content
+
+import (
+	"context"
+	"encoding/json"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras/internal/docker"
+)
+
+// Successors returns the nodes directly pointed by the current node, as well as
+// subject and config descriptor if applicable.
+func Successors(ctx context.Context, fetcher content.Fetcher, node ocispec.Descriptor) (nodes []ocispec.Descriptor, subject ocispec.Descriptor, config ocispec.Descriptor, err error) {
```

Comment:
If we return a pointer, we still need to do `config != nil && content.Equal(*config, desc)`

If we return a struct, we just need to do `content.Equal(config, desc)` since desc should not be a empty descriptor

The latter is cleaner in code.

Reply:
changed both config and subject to pointer

Reply:
changed both to pointers